solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -1839,7 +1839,7 @@ public DocCollection fetchCollectionState(String coll, Watcher watcher, String p
     while (true) {
       ClusterState.initReplicaStateProvider(() -> {
         try {
-          CommonTestInjection.injectDelay(); //for unit testing
+          CommonTestInjection.injectWaitForCondition(ZkStateReader.class.getName()); //for unit testing
```

Comment:
See comment regarding having this method unconditionally return `true` -- then we can invoke here as:
```java
assert CommonTestInjection.injectWaitForCondition(ZkStateReader.class.getName()); //for unit testing
```